chapters/feature-selection.qmd

```diff
@@ -0,0 +1,525 @@
+  geom_point(aes(col = `Automatic Selection`)) +
+  geom_line(aes(col = `Automatic Selection`)) +
+  facet_wrap(~label) +
+  labs(x = "# Noise Features", y = "RMSE Increase") +
+  scale_y_continuous(labels = scales::percent)
+```
+
+The results show that a few models, K-nearest neighbors, neural networks, and support vector machines, have severely degraded performance as the number of predictors increases. This is most likely due to the use of cross- and/or dot-products of the predictor columns in their calculations. The extra predictors add significant noise to these calculations, and that noise propagates in a way that inhibits the models from accurately determining the underlying relationship with the outcome. 
+
+However, several other models use these same calculations without being drastically affected. This is due to the models **automatically performing feature selection** during model training. Let’s briefly describe two clusters of models that were largely unaffected by the extra columns. 
+
+ 1. Penalized linear regression, generalized additive models, and RuleFit add a penalty to their calculations, restricting the model parameters from becoming abnormally large unless the underlying data warrant a large coefficient. This effectively reduces the impact of extra predictors by keeping their corresponding model coefficient at or near zero. 
+
+ 2. MARS and Cubist selectively include predictors in their regression equations, only including those specifically selected as having value. 
+
+The other unaffected models are tree-based. As seen in @fig-reg-tree, these types of models make different rules by selecting predictors to _split_ the data. If a predictor was not used in any split, it is functionally independent of the outcome in the prediction equation. This provides some insensitivity to the presence of non-informative predictors, although there is slight degradation in RMSE as the number of columns increases. 
+
+In summary, irrelevant columns only slightly affect ML models that automatically select features. Models that do not have this attribute can be crippled under the same circumstances. 
+
+## Different Levels of Features {#sec-feature-levels}
+
+When we talk about selecting features, we should be clear about what level of computation we mean. There are often two levels: 
+
+- **Original predictors** are the unmodified version of the data. 
+- **Derived predictors** refer to the set of columns that are present after feature engineering. 
+
+For example, if our model requires all numeric inputs, a categorical predictor is often converted to binary indicators. A categorical column for the day of the week is the original predictor, while the binary indicators for individual days are the derived predictor. Similarly, the 550 measurement predictors in the barley data are the original predictors, and embedded values, such as the PCA components, are the derived predictors. 
```

Comment:
```suggestion
For example, if our model requires all numeric inputs, a categorical predictor is often converted to binary indicators. A date-time column is the original predictor, while the binary indicators for individual days of the week are the derived predictor. Similarly, the 550 measurement predictors in the barley data are the original predictors, and embedded values, such as the PCA components, are the derived predictors. 
```

feels like it drives the point home more. Especially since you rarely would have a "day of the week" predictor that wasn't created by a date variable anyways